### DISCUSSION TITLE
Adding a Standard Library tutorial to Learning gem5

### DISCUSSION THREAD

**Discussion by jonathanhaley** [gem5/gem5#1675] [gem5-dev]
I was working through the learning gem5 tutorial (specifically the "Using the default configuration scripts" section) and discovered that I could not directly follow the tutorial as the se.py file had been deprecated. A couple of gem5 Developers Meetings ago, discussion took place about encouraging users to write python code using the standard library, as opposed to modifying the default configuration scripts using command line arguments. Therefore, I'm planning to write an update for this section of the website to instead discuss using the standard library and adding command line parameters exclusively for modification in the area of focus.
I figured it might be wise to open a discussion topic first to gather thoughts before proceeding.
Attached is an example script that could be used for this purpose.
[standard_library.txt](https://github.com/user-attachments/files/17376038/standard_library.txt)

**Comment by lsj134679** [comment]
I agree with your idea very much. Additionally, I would like to consult you; have you ever considered writing a similar script for fs mode? I once tried to enable fs mode with the ISA being RISC-V. However, when it comes to network topology, I wanted to enable Mesh, and I encountered some issues. The official sources do not provide relevant configuration scripts for the RISC-V architecture. If you could provide one, I think it would be greatly helpful to many beginners.
